crates/proto/src/error.rs

```diff
@@ -189,6 +189,9 @@ pub enum ProtoErrorKind {
         query: Box<Query>,
         /// If an SOA is present, then this is an authoritative response or a referral to another nameserver, see the negative_type field.
         soa: Option<Box<Record<SOA>>>,
+        /// Nameservers may be present in addition to or in lieu of an SOA for a referral
+        /// The tuple struct layout is vec[(Nameserver, [vec of glue records])]
+        ns: Option<Vec<(Record, Vec<Record>)>>,
```

Comment:
This crosses the threshold for me for something needs a named field type. I think it makes sense to name the fields in `(Record, Vec<Record>)` so that we can pass around `Option<Vec<NewThing>>`?

Reply:
I turned this into ForwardNSData {
 ns: Record
 glue: Vec<Record>
}